openedx/core/djangoapps/user_api/api.py

```diff
@@ -472,6 +475,88 @@ def _add_year_of_birth_field(self, form_desc, required=True):
             required=required
         )
 
+    def _add_profession_field(self, form_desc, required=True):
+        """Add a profession field to a form description.
+
+        Arguments:
+            form_desc: A form description
+
+        Keyword Arguments:
+            required (bool): Whether this field is required; defaults to True
+
+        """
+        # Translators: This label appears above a dropdown menu on the registration
+        # form used to select the user's profession
+        profession_label = _(u"Profession")
+
+        profession_options = configuration_helpers.get_value('PROFESSION_OPTIONS')
+        if profession_options is None:
+            # if there are no PROFESSION_OPTIONS set for this site make it a text box
+            form_desc.add_field(
+                "profession",
+                label=profession_label,
+                field_type="text",
+                include_default_option=False,
+                required=required,
+                error_messages={
+                    "required": accounts.REQUIRED_FIELD_PROFESSION_TEXT_MSG
+                }
+            )
+        else:
+            options = [(unicode(profession.lower()), ugettext_noop(profession)) for profession in profession_options]
```

Comment:
The if-else is only needed for the `options` variable. The call to `form_desc.add_field()` does not need to be duplicated.